tests/s2n_test.h

```diff
@@ -48,6 +56,13 @@ int test_count;
     EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(true));  \
     S2N_TEST_OPTIONALLY_ENABLE_FIPS_MODE();                    \
     EXPECT_SUCCESS_WITHOUT_COUNT(s2n_init());                  \
+    if (s2n_pq_is_enabled()) {                                 \
+        TEST_ALL = "test_all";                                 \
+        TEST_ALL_TLS12 = "test_all_tls12";                     \
+    } else {                                                   \
+        TEST_ALL = "test_all_no_pq";                           \
+        TEST_ALL_TLS12 = "test_all_tls12_no_pq";               \
+    }                                                          \
```

Comment:
Just a thought, but rather than needing to do this (and having separate security policies), would it make more sense to just ignore PQ ciphers during negotiation and consider kem preferences to be unset if NO_PQ was used / s2n_pq_enabled is false?

Reply:
I had considered that when I first started making these changes. @alexw91 and I talked about it and decided to go with this approach. While it makes testing patterns like this a bit awkward, the practice of failing fast and loud when attempting to do something that is disallowed (e.g. setting a PQ-aware security policy when PQ is disabled) seems most in line with s2n's practices of having the code/behavior be explicit and without surprising side-effects.

I can imagine trouble arising when consumers think they're negotiating PQ, but somehow `S2N_NO_PQ` was set and now the security policy is silently falling back to EC/DHE. Because of the fundamental differences between PQ and classic, that scenario seems distinct from something like: I thought I was negotiating x25519, but it wasn't supported so we were silently falling back to p256.

Of course it's open to discussion, so I'm interested if you have a strong opinion!

Reply:
That sounds like solid reasoning, carry on :)